File: problems_001to025/problem_025/solution.py
import os
import sys

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = 2

        a, b = 1, 1

        while len(str(b)) < 1000:
            a, b = b, a+b
            answer += 1

        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_001to025/problem_025/solution.ts
import { SolutionFunction } from '@/lib/types';

const solution: SolutionFunction = () => {
  let answer: ReturnType<SolutionFunction> = 2;

  let [a, b] = [1n, 1n];

  while (String(b).length < 1000) {
    [a, b] = [b, (a + b)];
    answer += 1;
  }

  return answer;
};

export default solution;
